test/unit/app/models/statistics_announcement_test.rb

```diff
@@ -292,6 +270,107 @@ class StatisticsAnnouncementTest < ActiveSupport::TestCase
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+  end
+
+  test "should allow updating to same publication type with attached draft publication" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::NationalStatistics.id,
+      title: "New title",
+    )
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should allow updating to different publication type to attached publication and updates the draft publication" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::OfficialStatistics.id,
+      title: "New title",
+    )
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+    assert_equal PublicationType::OfficialStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should not allow creating new announcement with publication of mismatched Publication type" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = build(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::OfficialStatistics.id,
+      title: "New title",
+    )
+
+    assert_not announcement.save
+    assert announcement.errors[:publication].include? "type does not match announcement type: must be 'Official Statistics'"
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should not allow updating Publication type with attached published publication" do
+    national_statistics = create(:published_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::OfficialStatistics.id,
+      title: "New title",
+    )
+
+    assert_not announcement.save
+    assert announcement.errors[:publication].include? "type cannot be modified when edition is in the published state"
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
```

Comment:
The test is about the type of the announcement, so maybe check that too/instead.

Reply:
the announcement failed to save with `assert_not announcement.save`, so the assumption there is nothing's changed - but i can add that assertion. 
but we shouldn't do instead because the point of the test is to test that should the validation fail and announcement fails to save - it should also not save the publication, hence publication type for the connected document needs to be the same

Reply:
My thinking was the same, just to show that the type of the announcement has stayed the same. Going back to the test namings... the test is about the announcement type not the publication's type. Or at least that's how I am reading these. There is one test that says "and updates the draft publication" - that's the only one that calls out that we care about the publication type.

Reply:
sure - i can add the publication check to the test name